includes/advanced-settings/class-accessibility-statement-page.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * Newspack Accessibility Statement Page functionality.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+/**
+ * Accessibility Statement Page class.
+ */
+class Accessibility_Statement_Page {
+
+	/**
+	 * Add hooks.
+	 */
+	public static function init() {
+		// Register REST routes.
+		add_action( 'rest_api_init', [ __CLASS__, 'register_rest_routes' ] );
+		// Add post status to accessibility statement page.
+		add_filter( 'display_post_states', [ __CLASS__, 'accessibility_post_status' ], 10, 2 );
+	}
+
+	/**
+	 * Create an accessibility statement page.
+	 *
+	 * @return array|null|WP_Error The created page data, null if it has been trashed or deleted, or error.
+	 */
+	public static function create_accessibility_statement_page() {
+		// Check if page ID is stored in theme_mods.
+		$page_id = get_theme_mod( 'accessibility_statement_page_id' );
+		if ( $page_id ) {
+			$page = get_post( $page_id );
+			// Check if page exists and is either published or draft.
+			if ( $page && 'page' === $page->post_type && in_array( get_post_status( $page->ID ), [ 'publish', 'draft' ] ) ) {
+				$page_data = [
+					'editUrl' => get_edit_post_link( $page->ID, 'raw' ),
+					'status'  => get_post_status( $page->ID ),
+					'pageUrl' => get_permalink( $page->ID ),
+				];
+				return $page_data;
+			}
+			// If page doesn't exist or is in trash, return null but keep the ID stored.
+			return null;
+		}
+
+		// If no page ID is stored, create a new page.
+		$page_id = wp_insert_post(
+			[
+				'post_title'   => __( 'Accessibility Statement', 'newspack-plugin' ),
+				'post_name'    => 'accessibility-statement',
+				'post_status'  => 'draft',
+				'post_type'    => 'page',
+				'post_content' => require NEWSPACK_ABSPATH . 'includes/advanced-settings/class-accessibility-statement-boilerplate.php',
```

Comment:
I think this would look better if we use output buffer. Instead of returning a long string, let's just have the file print its contents normally ([example](https://github.com/Automattic/newspack-plugin/blob/c5953294b16df0ed813796b52bfceb238c3442f8/includes/plugins/wc-memberships/block-patterns/pay-wall-one-tier.php)), and get it like this:

https://github.com/Automattic/newspack-plugin/blob/c5953294b16df0ed813796b52bfceb238c3442f8/includes/plugins/wc-memberships/class-block-patterns.php#L83-L85

Reply:
Ah, thanks @miguelpeixe! This has been updated -- it's much nicer 🙂